### PR TITLE
reduce tech debt and try to fix #94

### LINKED ISSUE #94: sandcastle: remove PVs even when a traceback happens
otherwise they linger and eat our quota

this is just a guess what's happening to be honest

### PR DESCRIPTION
**Pull request by TomasTomecek**
Related #94

\+ refactor the hell out of handlers and events

### CODE REVIEW

File: packit_service/worker/handler.py [hunk 1]
```diff
@@ -140,9 +141,10 @@ class JobHandler:
     name: JobType
     triggers: List[JobTriggerType]
 
-    def __init__(self, config: Config, job: JobConfig):
+    def __init__(self, config: Config, job: JobConfig, event: Event):
         self.config: Config = config
         self.job: JobConfig = job
+        self.event = event
```
Comment (rpitonak) [inline packit_service/worker/handler.py]: +1, can't understand why I didn't do it then.

Comment (TomasTomecek) [reply packit_service/worker/handler.py]: no worries, tech debt is the responsibility of all of us :)

**Comment by TomasTomecek** [comment]
I had to rewrite every single super() call once again b/c of the rebase, no more...

**Comment by TomasTomecek** [comment]
hups, forgot about pre-commit, installed it this time, so all should pass now

**Pull request merged**